OpenSim/Examples/DataTable/example.cpp

```diff
@@ -329,6 +369,145 @@ int main() {
     dt_vec3.clearData();
     dt_vec6.clearData();
 
+    // TimeSeriesDataTable_ is a DataTable_ that adds support for a timestamp
+    // column that can be used to index rows.
+    // Useful type aliases. The first template argument is the same as the one
+    // going into DataTable_ -- it is the type of the entries in the matrix. The
+    // second template argument is the type of the timestamp column.
+    using TsDataTableReal = OpenSim::TimeSeriesDataTable_<SimTK::Real, float>;
+    using TsDataTableVec3 = OpenSim::TimeSeriesDataTable_<SimTK::Vec3, float>;
+    using TsDataTableVec6 = OpenSim::TimeSeriesDataTable_<SimTK::Vec6, float>;
+
+    // Constructing a TimeSeriesDataTable_ is same as constructing a DataTable_.
+    // Timestamps can be added later to the timestamp column. The below code
+    // constructs a TimeSeriesDataTable_ of size 5x3 where all entries are
+    // filled will a given value.
+    TsDataTableReal tsdt_real{5, 3, 10};
+    TsDataTableVec3 tsdt_vec3{5, 3, SimTK::Vec3{10, 20, 30}};
+    TsDataTableVec6 tsdt_vec6{5, 3, SimTK::Vec6{10, 20, 30, 40, 50, 60}};
+
+    // Add timestamps to the timestamp column. Timestamp column must be
+    // increasing and before it is accessed, its length must equal the number of
+    // rows. Timestamps need not be added all at once but all rows must have an
+    // associated timestamp before timestamp column is used.
+    // The below code adds 5 timestamps, one for each row. Note the use of
+    // "0.0" and "1.0" instead of just "0" and "1". This is required so that
+    // all the entries in the list are deduced to be of the same type.
+    // Also note that type of the timestamp column and type of the entries in 
+    // the matrix can be different and they are in our examples -- Timestamp
+    // column entries are of type float and matrix entries are of type
+    // SimTK::Real, SimTK::Vec3 and SimTK::Vec6.
+    tsdt_real.addTimestamps({0.0, 0.25, 0.50, 0.75, 1.0});
+    tsdt_vec3.addTimestamps({0.0, 0.25, 0.50, 0.75, 1.0});
+    tsdt_vec6.addTimestamps({0.0, 0.25, 0.50, 0.75, 1.0});
+
+    // Both a timestamp and a row can be added in one call. Multiple timestamps
+    // and rows can be added in one call as well using addTimestampsAndRows().
+    data_real.clear();
+    data_vec3.clear();
+    data_vec6.clear();
+    data_real.assign(3, 20);
+    data_vec3.assign(3, SimTK::Vec3{40, 50, 60});
+    data_vec6.assign(3, SimTK::Vec6{70, 80, 90, 100, 110, 120});
+    tsdt_real.addTimestampAndRow(1.25, data_real);
```

Comment:
I think this command will be used more than anything else, so it'd be nice if the name were shorter. Could this just be an overload of `addRow()`?